gollm/azopenai.go

```diff
@@ -32,7 +32,7 @@ func init() {
 }
 
 func azureOpenAIFactory(ctx context.Context, u *url.URL) (Client, error) {
```

Comment:
Not quite sure why factory needs to receive a pointer to the URL. I would expect a copy to be enough for the clients.
https://github.com/GoogleCloudPlatform/kubectl-ai/blob/b5f05da496cde0cb1c159e1e31d770c5b7f135f2/gollm/factory.go#L39